lucille-core/src/test/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnectorTest.java

```diff
@@ -345,6 +346,115 @@ public void testJoiningDatabaseConnector() throws Exception {
+    configValues.put("jdbcUser", "");
+    configValues.put("jdbcPassword", "");
+    configValues.put("sql", "select name,type from animal order by name");
+    configValues.put("idField", "name");
+    // a list of other sql statements
+    ArrayList<String> otherSql = new ArrayList<>();
+    otherSql.add("select id as network_id,name,friendsWith from network order by name");
+    configValues.put("otherSQLs", otherSql);
+    // The join fields. id goes to animal_id
+    ArrayList<String> otherJoinFields = new ArrayList<>();
+    otherJoinFields.add("name");
+    configValues.put("otherJoinFields", otherJoinFields);
+    // create a config object off that map
+    Config config = ConfigFactory.parseMap(configValues);
+
+    // create the connector with the config
+    DatabaseConnector connector = new DatabaseConnector(config);
+    // run the connector
+    connector.execute(publisher);
+
+    List<Document> docs = messenger.getDocsSentForProcessing();
+    assertEquals(3, docs.size());
+    String expected = "{\"id\":\"Matt\",\"name\":\"Matt\",\"type\":\"Human\",\".children\":[{\"id\":\"0\",\"network_id\":3,\"name\":\"Matt\",\"friendswith\":\"Bob\"},{\"id\":\"1\",\"network_id\":4,\"name\":\"Matt\",\"friendswith\":\"Sonny\"},{\"id\":\"2\",\"network_id\":5,\"name\":\"Matt\",\"friendswith\":\"Blaze\"}],\"run_id\":\"testRunId\"}";
+    assertEquals(expected, docs.get(1).toString());
+
+    connector.close();
+    assertEquals(1, dbHelper.checkNumConnections());
+  }
+
+  @Test
+  public void testJoiningDatabaseConnectorNonComparable() throws Exception {
+    assertEquals(1, dbHelper.checkNumConnections());
+
+    HashMap<String, Object> configValues = new HashMap<>();
+    configValues.put("name", connectorName);
+    configValues.put("pipeline", pipelineName);
+    configValues.put("driver", "org.h2.Driver");
+    configValues.put("connectionString", "jdbc:h2:mem:test");
+    configValues.put("jdbcUser", "");
+    configValues.put("jdbcPassword", "");
+    configValues.put("sql", "select name,type from animal order by name");
+    configValues.put("idField", "name");
+    // a list of other sql statements
+    ArrayList<String> otherSql = new ArrayList<>();
+    otherSql.add("select id as _id,name,metadata from nonComparable order by name");
+    configValues.put("otherSQLs", otherSql);
+    // The join fields. id goes to animal_id
+    ArrayList<String> otherJoinFields = new ArrayList<>();
+    otherJoinFields.add("metadata");
+    configValues.put("otherJoinFields", otherJoinFields);
+    // create a config object off that map
+    Config config = ConfigFactory.parseMap(configValues);
+
+    // create the connector with the config
+    DatabaseConnector connector = new DatabaseConnector(config);
+    // run the connector, expect to throw non-comparable error
+    assertThrows("Other join field or joinId is not comparable.",ConnectorException.class, () -> connector.execute(publisher));

```

Comment:
Where is this exception/text in the code? I can't seem to find it.

Reply:
Somehow, the test still passes when error thrown does not have an accompanying message or a different message. Removed the exception text from the test in [91eec79](https://github.com/kmwtechnology/lucille/pull/184/commits/91eec79718ff3de576700b2d2911d5c1515ea985) as we are longer throwing an error with that message.